design/src/rfcs/rfc0007_timeout_config.md

```diff
@@ -0,0 +1,150 @@
+- **Middleware**: a term with several meanings,
+  - Generically speaking, middlewares are similar to Services and Layers in that they modify requests and responses.
+  - In the SDK, Middlewares have the specific meaning of being the only kind of struct that contains a `DispatchService`.
+  - Examples of Middleware include [MapRequest], [AsyncMapRequest], and [ParseResponse]. **TODO** is this true or am I getting things mixed up?
+- **DispatchService**: Connects Operation driven middleware to an HTTP implementation. It will also wrap the error type in OperationError to enable operation middleware reporting specific errors.
+  - **TODO** This definition is taken directly from a code comment. It makes it sound like this is the first and last layer that gets hit when making a request; Is that actually the case?
+- **Connector**: a term with several meanings,
+  - (esp. `DynConnector`) generally refers to a service that handles HTTP. HTTP is the lowest level we handle
+    - Dyn means that we're erasing the inner type
+    - **TODO** is this then a Service that we've erased the type of?
+  - A term from `hyper` for any object that implements the [Connect] trait. Actually just an alias for [tower_service::Service]. Sometimes referred to as a `Connection`.
+- **Stage**: A form of middleware that's not related to `tower`. These currently function as a way of transforming requests. We don't want to implement Timeouts as stages because we don't want to give them the ability to handle responses. This is because it would introduce lots of new complexity that we aren't yet ready to take on but this may change in the future.
+- **Stack**: higher order abstraction over Layers defined in the [tower crate][tower::layer::util::Stack] e.g. Layers wrap services in one another and Stacks wrap layers within one another.
+
+### Timeout terms
+
+- **Connect Timeout**: A limit on the amount of time after making an initial connect attempt on a socket to complete the
+  connect-handshake.
+    - _TODO: the runtime is based on Hyper which reuses connection and doesn't currently have a way of guaranteeing that
+      a fresh connection will be use for a given request._
+- **TLS Negotiation Timeout**: A limit on the amount of time a TLS handshake takes from when the CLIENT HELLO message is
+  sent to the time the client and server have fully negotiated ciphers and exchanged keys.
+    - _QUESTION: Our HTTP client supports multiple TLS implementations. Is there any reason that that would make
+      implementing this feature difficult?_
+- **Time to First Byte Timeout**: A limit on the amount of time an application takes to attempt to read the first byte over
```

Comment:
I thing a `write timeout` refers to the send-side equivalent if time to first byte